integration-tests/tests/src/tests/mixed.ts

```diff
@@ -220,23 +345,882 @@ describe("Mixed", () => {
+
+            // Create the objects that should pass the query string filter.
+            for (let count = 0; count < expectedFilteredCount; count++) {
+              this.realm.create(MixedSchema.name, { value: mixedDictionary });
+            }
+          });
+          const objects = this.realm.objects(MixedSchema.name);
+          expect(objects.length).equals(expectedFilteredCount + 2);
+
+          const insertedValues = Object.values(mixedDictionary);
+
+          for (const key in mixedDictionary) {
+            const valueToMatch = mixedDictionary[key];
+
+            // Objects with a dictionary value that matches the `valueToMatch` at the GIVEN key.
+            let filtered = objects.filtered(`value['${key}'] == $0`, valueToMatch);
+            expect(filtered.length).equals(expectedFilteredCount);
+
+            filtered = objects.filtered(`value['${key}'] == $0`, nonExistentValue);
+            expect(filtered.length).equals(0);
+
+            filtered = objects.filtered(`value['${nonExistentKey}'] == $0`, valueToMatch);
+            expect(filtered.length).equals(0);
+
+            filtered = objects.filtered(`value.${key} == $0`, valueToMatch);
+            expect(filtered.length).equals(expectedFilteredCount);
+
+            filtered = objects.filtered(`value.${key} == $0`, nonExistentValue);
+            expect(filtered.length).equals(0);
+
+            filtered = objects.filtered(`value.${nonExistentKey} == $0`, valueToMatch);
+            expect(filtered.length).equals(0);
+
+            // Objects with a dictionary value that matches the `valueToMatch` at ANY key.
+            filtered = objects.filtered(`value[*] == $0`, valueToMatch);
+            expect(filtered.length).equals(expectedFilteredCount);
+
+            filtered = objects.filtered(`value[*] == $0`, nonExistentValue);
+            expect(filtered.length).equals(0);
+
+            // Objects with a dictionary containing a key that matches `key`.
+            filtered = objects.filtered(`value.@keys == $0`, key);
+            expect(filtered.length).equals(expectedFilteredCount);
+
+            filtered = objects.filtered(`value.@keys == $0`, nonExistentKey);
+            expect(filtered.length).equals(0);
+
+            // Objects with a dictionary with the key `key` matching any of the values inserted.
+            filtered = objects.filtered(`value.${key} IN $0`, insertedValues);
+            expect(filtered.length).equals(expectedFilteredCount);
+
+            filtered = objects.filtered(`value.${key} IN $0`, [nonExistentValue]);
+            expect(filtered.length).equals(0);
+          }
+        });
+      });
+
+      describe("Notifications", () => {
+        let runCount = 0;
+
+        let testObjectWithList: TestObject;
+        let testObjectWithDictionary: TestObject;
+        let testList: Realm.List<any>;
+        let testDictionary: Realm.Dictionary<any>;
```

Comment:
Would it make sense to use the test context (setting and getting values on `this`) instead?

Reply:
See [this comment](https://github.com/realm/realm-js/pull/6364#discussion_r1471222487) 👍